addons/web/static/tests/fields/basic_fields_tests.js

```diff
@@ -1841,6 +1842,72 @@ QUnit.module('basic_fields', {
     });
 
+    QUnit.module('FieldBinarySignature');
+
+    QUnit.test('binary fields with Signature widget are correctly rendered', function (assert) {
+        assert.expect(8);
+
+        var form = createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch: '<form string="Partners">' +
+                    '<field name="signature" widget="signature"/>' +
+                '</form>',
+            res_id: 1,
+        });
+        assert.containsOnce(form, '.o_field_widget[name="signature"] .o_signature_empty',
+            "the empty Signature binary field should be rendered as a signature empty");
+
+        testUtils.form.clickEdit(form);
+
+        assert.containsOnce(form, '.o_field_widget[name="signature"] .o_signature_empty',
+        "the empty Signature binary field should be rendered as a signature empty in edit mode");
+
+        testUtils.dom.click(form.$('.o_field_widget[name="signature"] > .o_signature'));
+
+        assert.strictEqual($('.modal').length, 1, "should have opened the dialog");
+
+        assert.containsOnce($, '.modal-content .o_sign_clean', "there shoud have a button to clear the signature");
+
+        assert.strictEqual($('.o_sign_mode_draw:visible').length, 1, "there shoud have a button to draw the signature");
+        assert.strictEqual($('.o_sign_mode_auto:visible').length, 0, "there shoud not have button to auto draw the signature");
+        assert.strictEqual($('.o_sign_mode_load:visible').length, 1, "there shoud have a button to load the signature");
+
+        assert.strictEqual($('.jSignature:visible').length, 1, "there shoud have a canvas to draw the signature");
+        form.destroy();
+    });
+
+    QUnit.test('binary fields with Signature widget with option for name are correctly rendered', function (assert) {
+        assert.expect(5);
+
+        var form = createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch: '<form string="Partners">' +
+                    '<field name="display_name"/>' +
+                    '<field name="signature" widget="signature" options="{\'full_name\': \'display_name\'}"/>' +
+                '</form>',
+            res_id: 2,
+        });
+
+        testUtils.form.clickEdit(form);
+
+        testUtils.dom.click(form.$('.o_field_widget[name="signature"] > .o_signature'));
+
+        assert.strictEqual($('.o_sign_mode_draw:visible').length, 1, "there shoud have a button to draw the signature");
+        assert.strictEqual($('.o_sign_mode_auto:visible').length, 1, "there shoud have a button to auto draw the signature");
+        assert.strictEqual($('.o_sign_mode_load:visible').length, 1, "there shoud have a button to load the signature");
+
+        assert.containsOnce($, '.o_sign_font_dialog', "there shoud have a panel with fonts");
+        assert.ok($('.o_sign_font_selection > a').length > 2, "should have more than 2 fonts");
```

Comment:
this is a bad assert: 'more than 2' is not specific, you control exactly how many to you can be specific, and use assert.containsN with that specific number